voter/utils.py

```diff
@@ -0,0 +1,128 @@
+from datetime import datetime, timezone
+from enum import Enum
+import os
+import subprocess
+
+import requests
+
+from voter.models import FileTracker
+
+
+FETCH_STATUS_CODES = Enum("FETCH_STATUS_CODES",
+                          "CODE_OK CODE_NET_FAILURE CODE_WRITE_FAILURE CODE_NOTHING_TO_DO")
+
+
+def get_output_file(output):
+    """
+    If we're running in no-output mode, then return a handle to /dev/null to be
+    used as the `file` argument to `print` statements. Otherwise return None,
+    which tells `print` to write to stdout.
+    """
+    if not output:
+        return open(os.devnull, 'w')
+
+
+def tqdm_or_quiet(output):
+    """Get our progress-bar generator OR a no-op if we're running in no-output mode."""
+    if output:  # pragma: no cover
+        from tqdm import tqdm
+    else:
+        def tqdm(x, **kw):
+            return x
+    return tqdm
+
+
+def derive_target_folder(base_path, now):
+    now_str = now.strftime("%Y-%m-%dT%H:%M:%S:%s")
```

Comment:
Is this a good time to remove the colon from these paths like we originally said we should do?